UI/MapWnd.cpp

```diff
@@ -1939,26 +1939,94 @@ void MapWnd::RenderSystems() {
                     if (system->NumStarlanes() > 0) {
-                        glColor(GetOptionsDB().Get<GG::Clr>("UI.unowned-starlane-colour"));
 
+                        bool has_empire_planet = false;
+                        bool has_neutrals = false;
+                        std::map<int, int> colony_count_by_empire_id;
+                        const std::set<int>& known_destroyed_object_ids = GetUniverse().EmpireKnownDestroyedObjectIDs(HumanClientApp::GetApp()->EmpireID());
+
+                        for (std::shared_ptr<const Planet> planet : Objects().FindObjects<const Planet>(system->PlanetIDs())) {
+                            if (known_destroyed_object_ids.count(planet->ID()) > 0)
+                                continue;
+
+                            // remember if this system has a player-owned planet, count # of colonies for each empire
+                            if (!planet->Unowned()) {
+                                has_empire_planet = true;
+
+                                std::map<int, int>::iterator it = colony_count_by_empire_id.find(planet->Owner()) ;
+                                if (it != colony_count_by_empire_id.end()) {
+                                    it->second++;
+                                } else {
+                                    colony_count_by_empire_id.insert({planet->Owner(), 1});
+                                }
+                            }
+
+                            // remember if this system has neutrals
+                            if (planet->Unowned() && !planet->SpeciesName().empty() && planet->CurrentMeterValue(METER_POPULATION) > 0.0) {
+                                has_neutrals = true;
+                            }
+                        }
+
+                        // draw inner circle in color of supplying empire
                         int empire_id = GetSupplyManager().EmpireThatCanSupplyAt(system_icon.first);
                         if (empire_id != ALL_EMPIRES) {
                             if (Empire* empire = GetEmpire(empire_id)) {
                                 glColor(empire->Color());
                             }
+                        } else {
+                            glColor(GetOptionsDB().Get<GG::Clr>("UI.unowned-starlane-colour"));
                         }
+
+                        glLineWidth(2.0f);
                         CircleArc(circle_ul, circle_lr, 0.0, TWO_PI, false);
+
+                        // systems with neutrals and no empire have a segmented outer circle
+                        if (has_neutrals && !(has_empire_planet)) {
+                            glLineWidth(2.0f);
+                            glColor(ClientUI::TextColor());
+
+                            float segment = static_cast<float>(TWO_PI) / 24.0f;
+                            for (int n = 0; n < 24; n = n + 2) {
+                                CircleArc(outer_circle_ul, outer_circle_lr, n * segment, (n+1) * segment, false);
```

Comment:
I think if you're rendering 12 arc bits, it should be doable with a single render call, rather than using this function.